### PR TITLE
Sending command to ip pipeline via telnet

### PR DESCRIPTION
**Pull request by PierluigiGreto**

### CODE REVIEW

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+ #!/usr/bin/env python3
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py]: Any reason for the whitespace

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 2]
```diff
@@ -18,6 +18,74 @@
 CONFIG['telnet_port_uplink'] = 8086
 CONFIG['telnet_host_downlink'] = '127.0.0.1'
 CONFIG['telnet_port_downlink'] = 8087
+gateway_address="00:e0:ed:5c:96:fe"
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py:21]: Is this the new GW MAC ?

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 3]
```diff
@@ -66,7 +134,7 @@ def main():
                     s = s.decode('utf8').replace("'", '"')
                     try:
                         d = json.loads(s)
-                        print(d)
+                        handle_message(d)
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py:135]: :+1:

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 4]
```diff
@@ -18,6 +18,74 @@
...
+    s_tag, c_tag = tags[1], tags[2]
+    command = f'pipeline downstream|firewall table 0 rule add match acl priority 0 ipv4 0.0.0.0 0 {d["ip_addr"]} 32 0 65535 0 65535 17 action fwd port 0'
+    send_telnet_command(direction, command)
+    command = f'pipeline downstream|firewall table 0 rule add match acl priority 0 ipv4 0.0.0.0 0 {d["ip_addr"]} 32 0 65535 0 65535 6 action fwd port 0'
+    send_telnet_command(direction, command)
+    last = d["ip_addr"].split(".")[3]
+    command = f'pipeline downstream|hqos table 0 rule add match lpm ipv4 {d["ip_addr"]} 32 action fwd port 0 tm subport 0 pipe {last}'
+    send_telnet_command(direction, command)
+    command = f'pipeline downstream|routing table 0 rule add match acl priority 0 ipv4 0.0.0.0 0 {d["ip_addr"]} 32 0 65535 0 65535 17 action fwd port 0 encap qinq_pppoe {d["calling_station_id"]} {d["called_station_id"]} 7 0 {s_tag} 7 0 {c_tag} {d["pppoe_sessionid"]}'
+    #command = f'pipeline downstream|routing table 0 rule add match acl priority 0 ipv4 0.0.0.0 0 {d["ip_addr"]} 32 0 65535 0 65535 17 action fwd port 0 encap ether {d["calling_station_id"]} {d["called_station_id"]}'
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py]: Do we need that comment ... Maybe add a line that says when we should uncomment

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 5]
```diff
@@ -18,6 +18,74 @@
...
+    command = f'pipeline upstream|routing table 0 rule add match acl priority 0 ipv4 {d["ip_addr"]} 32 0.0.0.0 0 0 65535 0 65535 6 action fwd port 0 encap ether {gateway_address} {d["called_station_id"]}'
+    send_telnet_command(direction, command)
+
+def send_downlink_message(d):
+    print("############################################ DOWNSTREAM ####################################################################")
+    direction = 'downlink'
+    ctrl_ifname = d['ctrl_ifname']
+    tags = ctrl_ifname.split(".")
+    if len(tags) != 3:
+        print("Error: the interface name is not as expected. es: eth1.0.37")
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py]: es = e.g. ? ... is this Italian :rofl:

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py [hunk 6]
```diff
@@ -18,6 +18,74 @@
...
+        send_uplink_message(d)
+        send_downlink_message(d)
+
+def send_uplink_message(d):
+    print("############################################ UPSTREAM ####################################################################")
+    direction = 'uplink'
+    ctrl_ifname = d['ctrl_ifname']
+    tags = ctrl_ifname.split(".")
+    if len(tags) != 3:
+        print("Error: the interface name is not as expected. es: eth1.0.37")
```
Comment (butjar) [inline dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py]: e.g. ?

**Review by butjar — COMMENTED** [review]
Looks good ... apply (or ignore) my suggestions, merge and test please

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py (hunk 5 above)
Comment (PierluigiGreto) [reply dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py]: Ahahah Yes

File: dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py (hunk 2 above)
Comment (PierluigiGreto) [reply dpdk-ip-pipeline-cli/dpdk-ip-pipeline-cli.py:21]: Yes

**Comment by butjar** [comment]
:ship:

**Pull request merged**